experimental/generation/generator/packages/library/src/dialogGenerator.ts

```diff
@@ -467,6 +467,13 @@ async function processTemplates(
     force: boolean,
     feedback: Feedback): Promise<void> {
     scope.templates = {}
+    // foreach property
+    //   foreach entity
+    //     foreach template
+    //       process template
+    // 1) property only template -> first time it get populated I've got it
+    // 2) property+entity template -> entity must be bound
+    // 3) property + fixed entity template -> will only get generated once
     for (let locale of locales) {
```

Comment:
// We expand each template in the context of a combination of property and entity.
// Since we only generate each filename once this means property only templates don't care about the entity and entity templates will be expanded for each entity.
// If a template author wants to do different things with different entities they will need explicit templates for each desired entity.